libafl_qemu/libafl_qemu_sys/src/x86_64_stub_bindings.rs

```diff
@@ -13703,7 +13702,7 @@ extern "C" {
 extern "C" {
     pub fn libafl_add_read_hook(
         gen: ::std::option::Option<
-            unsafe extern "C" fn(
+             extern "C" fn(
```

Comment:
Shouldn't this be unsafe? The raw pointer will later be dereferenced (or is the later function marked unsafe?)

Reply:
Maybe, but then we need to keep the other similar function signatures consistent.
In any case, here it's a problem since we were transmuting a safe function pointer to an unsafe one, which is not a good idea IMHO.
I think the rationale behind making it safe is that we are using it through the Qemu struct, which is unsafe by nature. I'd say that as long as the signature of the function pointer is correct, it should not cause additional safety issues.

FYI it's used concretely [there](https://github.com/AFLplusplus/qemu-libafl-bridge/blob/9f3e2399ee9b106dfbb8c3afcdfdf30e235fc88f/libafl/hook.c#L456-L457).

Reply:
IMHO a function that takes a `*mut` ptr and potentially dereferences that can never be safe. So I would argue the original function should already be unsafe (and then we don't transmute safe -> unsafe either)

Reply:
Ah lol nevermind it's C :P It's all unsafe anyway

Reply:
Sounds reasonable, it's true safe rust allows the creation of any raw pointer. It's better to properly propagate the unsafeness even for an unsafe structure.

Reply:
The good point is that we can remove the ugly fix of the bindings.